packages/core/src/internal/base/button.base.ts

```diff
@@ -154,5 +191,14 @@ export class CdsBaseButton extends LitElement {
       this.tabIndexAttr = this.disabled ? -1 : 0;
       this.ariaDisabled = this.disabled ? 'true' : 'false';
     }
+
+    // Forms handle default submit buttons without type="submit" automatically.
+    // Because of the dynamic createion of the native element, we prefer to set the attribute explicitly.
+    if (!this.type && this.closest('form')) {
```

Comment:
if we make `type` a getter/setter and store the actual value in a private property (a pattern we've used a lot in the angular components), then could this logic could live in the setter?

Reply:
I guess it won't get called if it's instantiated without type="something" attribute. But I'll test it.